### ISSUE TITLE
Problem with job deletion when using HA (High Availability)

### ISSUE THREAD

**Issue by shashken** [GoogleCloudPlatform/flink-on-k8s-operator#320]
**TLDR**
When a job cluster is deleted the job should be cancelled to allow the JM to delete the submitted job graph from HA storage (So it wont be loaded when resubmitting)

**What happends:**
Submit a job cluster with HA configured -> works
Delete it
Submit it again -> The previous job is loaded from HA, it gets cancelled by the operator and the new job is not working 

**Why it happends**
When deleting a job cluster with high availability (kubectl delete) the flink pods are terminated before the flink job is cancelled (This is the bug),
This will not delete the submitted job graph from the high availability storage.
When submitting the job cluster again it will look for an existing job, it will find the previous one as it was not cancelled, 
Now the operator will cancel the job as it does not know it (controllers/flink_reconciler.go:432) and the job starts as cancelled

**Comment by functicons** [comment]
Yep, in the current operator design, deleting a job cluster resource will automatically deleting its child resources including JM, TM and the job submitter, but it won't clean up states in external HA store.

The fix is not so simple, we need to think it through. As a workaround, you can try not reusing job cluster name for the new cluster.

**Comment by shravangit20** [comment]
@shashken @functicons Could you please share inputs on how to Submit a job cluster with HA configured? If possible could you please share your config?
